File: src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

import java.util.ArrayList;

/**
 *
 * @author juan
 */
public class FuzzyPlayer extends Player{
    
    public FuzzyPlayer(Player other){
        super(other);
    }
    
    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        Directions dir=super.move(direction, validMoves);
        return Dice.nextStep(dir, validMoves, this.getIntelligence());
    }
    
    @Override
    public float attack(){
        return (Dice.intensity(this.getStrength())+this.sumWeapons());
    }
    
    @Override
    protected float defensiveEnergy(){
        return (Dice.intensity(this.getIntelligence())+this.sumShields());
    }
    @Override
    public String toString(){
        String output="Fuzzy ";
        output+=super.toString();
        
        return output;
    }
}

File: FuzzyPlayer.rb
require_relative 'Player'


module Irrgarten

  class FuzzyPlayer < Player
    # Constructor de la clase {FuzzyPlayer}.
    def initialize(other_player)
        copy(other_player)
    end

    def move(direction, valid_moves)
      dir=super(direction,valid_moves)
      return Dice.next_step(dir, valid_moves,self.intelligence)
    end

    def attack
      return Dice.intensity(self.strength+self.sum_weapons)
    end

    protected

    def defensive_energy
      return Dice.intensity(self.intelligence+self.sum_shields)
    end

    public
    def to_s
      return "(Fuzzy) " + super
    end
  end
end


